lib/random.gd

```diff
@@ -172,6 +172,15 @@ DeclareOperation( "Random", [IsRandomSource, IsInt, IsInt] );
     BIND_GLOBAL("InstallOtherMethodWithRandomSource", func(InstallOtherMethod));
 end)();
 
+
+InstallMethodWithRandomSource( Random,
+    "for a random source and a (finite) collection",
+    [ IsRandomSource, IsCollection and IsFinite ],
+    {rs,C} -> RandomList(rs, Enumerator( C ) ) );
+
+RedispatchOnCondition(Random,true,[IsCollection],[IsFinite],0);
+RedispatchOnCondition(Random,true,[IsRandomSource, IsCollection],[IsFinite],0);
```

Comment:
Hm, I am not so happy about moving these collection specific functions out of `coll.gi`. Here is a possible alternative, which allows using `InstallMethodWithRandomSource` inside `coll.gi`:
```patch
diff --git a/lib/coll.gi b/lib/coll.gi
index 264a63de0..3f63919b2 100644
--- a/lib/coll.gi
+++ b/lib/coll.gi
@@ -246,11 +246,6 @@ InstallMethod( RepresentativeSmallest,
 ##  an enumerator of <C> and selects a random element of this list using the
 ##  function `RandomList', which is a pseudo random number generator.
 ##
-if IsHPCGAP then
-    MakeThreadLocal( "GlobalMersenneTwister" );
-else
-    DeclareGlobalVariable( "GlobalMersenneTwister" );
-fi;
 InstallGlobalFunction( RandomList, function(list)
   return list[Random(GlobalMersenneTwister, 1, Length(list))];
 end );
diff --git a/lib/random.gd b/lib/random.gd
index c582a930c..763c31482 100644
--- a/lib/random.gd
+++ b/lib/random.gd
@@ -39,7 +39,7 @@
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
-BindGlobal( "RandomSourcesFamily", NewFamily( "RandomSourcesFamily" ) );    
+BIND_GLOBAL( "RandomSourcesFamily", NewFamily( "RandomSourcesFamily" ) );
 DeclareCategory( "IsRandomSource", IsComponentObjectRep );
 
 #############################################################################
@@ -105,6 +105,14 @@ DeclareOperation( "Random", [IsRandomSource, IsInt, IsInt] );
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
+
+if IsHPCGAP then
+    MakeThreadLocal( "GlobalMersenneTwister" );
+else
+    DeclareGlobalVariable( "GlobalMersenneTwister" );
+fi;
+
+
 (function()
     local func;
     func := function(installType)
diff --git a/lib/read1.g b/lib/read1.g
index b8d21f9c8..a360f1c4b 100644
--- a/lib/read1.g
+++ b/lib/read1.g
@@ -36,6 +36,8 @@ ReadLib( "set.gd"      );
 
 ReadLib( "record.gd"   );
 
+ReadLib( "random.gd"   );
+
 ReadLib( "cache.gi"    );
 ReadLib( "coll.gi"     );
 
@@ -70,8 +72,6 @@ ReadLib( "info.gi"     );
 ReadLib( "assert.gi"   );
 ReadLib( "global.gi"   );
 
-ReadLib( "random.gd"   );
-
 ReadLib( "options.gd"  );
 ReadLib( "options.gi"  );
 
```

To me, it anyway nicer to declare `GlobalMersenneTwister` inside `random.gd` instead of `coll.gi` -- this matches `GlobalRandomSource`. The patch above is not quite what I'd do, though: I'd rather move the definition of `GlobalMersenneTwister` where it belongs, i.e. next to the def of `GlobalRandomSource` (where we actually already have commented out definitions for `GlobalMersenneTwister`!). But this means the def for `InstallMethodWithRandomSource` has to be moved downwards, which is simple, but leads to a huge diff).

I can also make a PR with these changes now...